python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -164,6 +174,14 @@ def test_sequence_types(seq):
 
 
 @parametrize_with_iterable_types
+def test_nested_sequence_types(seq):
+    arr1 = pa.array([seq([1, 2, 3])])
+    arr2 = pa.array([[1, 2, 3]])
+
+    assert arr1.equals(arr2)
```

Comment:
This is the new test that actually tests this feature, but all other tests that remained parametrised with `parametrize_with_iterable_types` trigger the new code path too.